### PR TITLE
[Spritelab] Update feedback dialog for final/freeplay levels

### PR DESCRIPTION
**Pull request by maddiedierker**
Updates the finish dialog title string and print option for SpriteLab (and Poetry since it inherits from SpriteLab) only. This only affects the finish dialog for freeplay levels that are the final level in a progression.

**Note for triage meeting:** This change will likely cause some Eyes test failures.

**Before:**
Screen Shot 2021-11-24 at 1 28 16 PM

**After:**
Updates the title and removes the print button.
Screen Shot 2021-11-24 at 9 31 25 AM

This change is tightly coupled to the "success" case of FeedbackUtils.prototype.getFeedbackMessage:

https://github.com/code-dot-org/code-dot-org/blob/6be244bafd471fe390edba08077acdcb0db55c9e/apps/src/feedback.js#L875-L914

I've added unit tests for that case in this PR to make sure it behaves how we expect. I made the code change (most of the diff in this PR is from unit tests) as minimal as possible to avoid changing behavior outside of Spritelab since we're so close to HoC.

## Links

- [STAR-1944](https://codedotorg.atlassian.net/browse/STAR-1944)

## Testing story

Adds unit tests for this scenario in feedback.js. This file (which is 2,000 lines and core to our progression logic) had **no** unit tests 😱 

## Follow-up Work
- [STAR-1976](https://codedotorg.atlassian.net/browse/STAR-1976)

### CODE REVIEW

File: apps/src/p5lab/P5Lab.js [hunk 1]
```diff
@@ -1572,12 +1594,13 @@ export default class P5Lab {
...
       appStrings: {
-        reinfFeedbackMsg: msg.reinfFeedbackMsg(),
+        reinfFeedbackMsg,
         sharingText: msg.shareGame()
       },
       hideXButton: true,
       saveToProjectGallery: saveToProjectGallery,
-      disableSaveToGallery: !isSignedIn
+      disableSaveToGallery: !isSignedIn,
+      disablePrinting: this.disablePrinting()
```
Comment (maddiedierker) [inline apps/src/p5lab/P5Lab.js:1603]: in our post-HoC follow-up work, this function will be changed to `false` (EDIT: `true` 🤦🏻‍♀️ ) to disable printing in all P5Labs. being so close to HoC, we only wanted to disable printing in Spritelab/Poetry for now so we can re-evaluate this configuration setting across all other labs later. it should likely only be enabled in Artist, so this setting is probably backwards (e.g., you should have to opt in, not opt out)

Comment (maddiedierker) [reply apps/src/p5lab/P5Lab.js:1603]: this dialog is the only place where this setting is consumed:

https://github.com/code-dot-org/code-dot-org/blob/6be244bafd471fe390edba08077acdcb0db55c9e/apps/src/templates/sharing.html.ejs#L33-L37

File: apps/src/p5lab/P5Lab.js [hunk 2]
```diff
@@ -1572,12 +1594,13 @@ export default class P5Lab {
       // feedbackImage: feedbackImageCanvas.canvas.toDataURL("image/png")
       showingSharing: !level.disableSharing && level.freePlay,
       appStrings: {
-        reinfFeedbackMsg: msg.reinfFeedbackMsg(),
+        reinfFeedbackMsg,
```
Comment (maddiedierker) [inline apps/src/p5lab/P5Lab.js:1597]: it's kind of linked in the PR description, but, more specifically, this setting/string is only used in this branch of FeedbackUtils.prototype.getFeedbackMessage:

https://github.com/code-dot-org/code-dot-org/blob/6be244bafd471fe390edba08077acdcb0db55c9e/apps/src/feedback.js#L891-L904

File: apps/src/p5lab/P5Lab.js (hunk 1 above)
Comment (epeach) [reply apps/src/p5lab/P5Lab.js:1603]: Howdy - maybe I'm not understanding - it sounds like we want to only disable printing in Spritelab/Poetry, but this.disablePrinting returns false, so will it disable printing across all P5Lab?

Comment (maddiedierker) [reply apps/src/p5lab/P5Lab.js:1603]: woops, my first comment was the opposite of what i wanted! i updated the comment, but also want to clarify here:

after HoC, we'll remove the `this.disablePrinting` function and update this configuration to `{disablePrinting: true}` for all P5Labs (and likely other labs)

**Review by epeach — APPROVED** [review]
Some really good clean-up in here and *claps* for adding testing where we needed some.

**Pull request merged**